### PR TITLE
[release-1.7] 🌱 Add verify-govulncheck target and integrate to scan action

### PR DESCRIPTION
**Pull request by chrischdi**
This is a manual cherry-pick of https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2174

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -370,6 +375,26 @@ verify-boilerplate: ## Verify boilerplate text exists in each file
...
+verify-security: ## Verify code and images for vulnerabilities
+	$(MAKE) verify-container-images && R1=$$? || R1=$$?; \
+	$(MAKE) verify-govulncheck && R2=$$? || R2=$$?; \
+	if [ "$$R1" -ne "0" ] || [ "$$R2" -ne "0" ]; then \
+	  echo "Check for vulnerabilities failed! There are vulnerabilities to be fixed"; \
+		exit 1; \
+	fi
+
+.PHONY: verify-flavors
+verify-flavors: $(FLAVOR_DIR) generate-flavors ## Verify generated flavors
```
Comment (sbueringer) [inline Makefile:392]: Looks like this is not part of the original cherry-pick.

@chrischdi Maybe for simplicity. Can you open PRs for all branches (including 1.8) to resync .github/workflows + the Makefile changes? (including the changes to the generate-flavors tool)

Comment (killianmuldoon) [reply Makefile:392]: Would that sync include the failed cherry-picks from https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2167 ? Don't want to hit those twice

Comment (sbueringer) [reply Makefile:392]: Yup. Talked to Christian. He'll do a series of cherry-picks later today to bring GitHub actions, Makefile and related scripts in sync across 1.8-1.5

Comment (killianmuldoon) [reply Makefile:392]: SGTM - I'd just started looking at the failing cherry-picks, luckily I saw this too :smile:

Comment (sbueringer) [reply Makefile:392]: Yup sorry missed writing this somewhere while debugging our CI env :)

**Pull request closed without merging**